### PR TITLE
fix(data): add proper JSON-API contentType header to the response

### LINKED ISSUE #141: Ensure responding with JSON-API content-type header is allowed
By default Denali responds with a content type of 'application/json' As mentioned in the spec, a JSON-API response content type must be: `Content-Type: application/vnd.api+json`

Looks like this can be tweaked at a config level (although there are some todos about the config in code).

Don't have time ATM to test this thoroughly but will aim to come back to this later ...

### PR DESCRIPTION
**Pull request by acorncom**
Fixes #141

### CODE REVIEW

File: test/unit/actions-test.js [hunk 1]
```diff
@@ -103,6 +103,32 @@ describe('Denali.Action', function() {
...
+  //     }
+  //   });
+  //   let action = new TestAction(mock);
+
+  //   return action.run().then(() => {
+  //     expect(responded).to.be.true();
+  //     expect(serialized).to.be.true();
+  //   });
+  // });
+
```
Comment (acorncom) [inline test/unit/actions-test.js]: This is quite ugly here.  I tried `skip('it should use')` and `this.skip()` inside the test, not sure what was going wrong here?

**Comment by acorncom** [comment]
@davewasmer @knownasilya @sivakumar-kailasam Thoughts on doing things this way?

**Comment by knownasilya** [comment]
Looks fine to me, but what if we need other headers in the future?

**Comment by acorncom** [comment]
Other headers per serializer or other headers for other serializers?

**Comment by knownasilya** [comment]
Both? lol

File: lib/runtime/action.js [hunk 2]
```diff
@@ -185,6 +185,7 @@ export default class Action {
       }
       let serializer = this.container.lookup(`serializer:${ type }`);
       response.body = serializer.serialize(response.body, assign({ action: this }, response.options));
+      response.options.contentType = serializer.contentType;
```
Comment (davewasmer) [inline lib/runtime/action.js]: I wonder if we should just pass the Response object to the serializer. It feels a bit more future proof (i.e. what if a serializer wants to add some additional headers, like Link headers for pagination or something. If it get's the entire Response object, you could theoretically tack on additional metadata to the Response, and have the serializer render that however it sees fit.

Comment (acorncom) [inline lib/runtime/action.js]: This sounds good to me.

Comment (acorncom) [inline lib/runtime/action.js]: @davewasmer You after something like this?

```js
      response = serializer.serialize(response, assign({ action: this }, response.options));
```

**Comment by knownasilya** [comment]
@davewasmer close this?

**Pull request closed without merging**